sherpa/astro/io/__init__.py

```diff
@@ -852,17 +1221,33 @@ def _remove_structural_keywords(header: HdrTypeArg) -> HdrType:
 
     """
 
-    out = {}
-    for key, value in header.items():
-        if value is None or _is_structural_keyword(key):
-            continue
+    return [item for item in header.values
+            if not _is_structural_keyword(item.name)]
 
-        out[key] = value
 
-    return out
+# By making this a package-level value, users can actually change the
+# text (and in fact the name) by using code like
+#
+#    from sherpa.astro.io import CREATOR
+#    CREATOR.value = "not sherpa"
+#
+# but it does not seem worth documenting this at this time.
```

Comment:
A couple of random notes here

1. I was actually thinking of you when I added this doc note!
2, are we convinced this is a sensible "API"; for instance I think that (should really check this) that if a user just creates a `"CREATOR"` keyword in the header of the dataset then this will get used instead, which might be a "better" way for users to control the output metadata
3. documenting this does mean we are "signing on" to keep this updated (e.g. your worries in recent discussion over whether we want to add some sort of API for changing plot labels via the data object)
4. note that this change here **only** sets the CREATOR keyword; support for `write/set_arf` and `write/set_rmf` were added in the 4.16 release so you should already be able to change your ray-tracing code.